cli/build.rs

```diff
@@ -1,7 +1,188 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+use deno_core::js_check;
+use deno_core::CoreOp;
+use deno_core::ErrBox;
+use deno_core::Isolate;
+use deno_core::Op;
+use deno_core::PinnedBuf;
+use deno_core::StartupData;
+use serde::Deserialize;
+use serde_json::json;
+use serde_json::Value;
 use std::env;
+use std::fs;
 use std::path::PathBuf;
 
+#[derive(Deserialize)]
+struct FetchAssetArgs {
+  name: String,
+}
+
+fn op_fetch_asset(
+  custom_assets: Vec<(String, PathBuf)>,
+  control: &[u8],
+) -> CoreOp {
+  let result = serde_json::from_slice(control)
+    .map_err(ErrBox::from)
+    .and_then(move |args| inner_fetch_asset(custom_assets, args));
+
+  let response = match result {
+    Ok(v) => json!({ "ok": v }),
+    Err(err) => json!({ "err": err.to_string() }),
+  };
+
```

Comment:
This file is so verbose and repetetive on purpose

Reply:
Also it's pretty cool demonstration how to use Isolates during build time 😄

Reply:
It seems the use of serde here isn't adding that much. What about returning the asset as a raw byte array?